ceno_emul/src/syscalls/keccak_permute.rs

```diff
@@ -55,22 +55,12 @@ impl From<KeccakState> for [Word; KECCAK_WORDS] {
 pub fn keccak_permute(vm: &VMState) -> SyscallEffects {
     let state_ptr = vm.peek_register(Platform::reg_arg0());
 
-    // for compatibility with sp1 spec
-    assert_eq!(vm.peek_register(Platform::reg_arg1()), 0);
-
     // Read the argument `state_ptr`.
-    let reg_ops = vec![
-        WriteOp::new_register_op(
-            Platform::reg_arg0(),
-            Change::new(state_ptr, state_ptr),
-            0, // Cycle set later in finalize().
-        ),
-        WriteOp::new_register_op(
```

Comment:
This is just to compatible with sp1 binary logic to assure x11 (a2) register are 0.
In Ceno we do not need this check, as our precompile circuit do not need this check